### ISSUE TITLE
[Fronius] Add channels for secondary and tertiary DC input

### ISSUE THREAD

**Issue by curlyel** [openhab/openhab-addons#12856]
The channels of the inverter thing for Fronius include DC current and DC voltage. Unfortunately, those channels just get the readings for the first DC input of the inverter. Readings **for the second DC input as well as for the battery input are not yet parsed**:

![image](https://user-images.githubusercontent.com/11849305/171468628-aaae57b6-9fc9-42a1-b3ca-8108f0046a61.png)

When I check for the json provided by the `solar_api`:

`http://<inverter-ip>/solar_api/v1/GetInverterRealtimeData.cgi?Scope=Device&DeviceId=1&DataCollection=CommonInverterData`

I get this response json:

```
{
   "Body" : {
      "Data" : {
         "DAY_ENERGY" : {
            "Unit" : "Wh",
            "Value" : null
         },
         "DeviceStatus" : {
            "ErrorCode" : 0,
            "InverterState" : "Running",
            "StatusCode" : 7
         },
         "FAC" : {
            "Unit" : "Hz",
            "Value" : 49.982288360595703
         },
         "IAC" : {
            "Unit" : "A",
            "Value" : 2.8039026260375977
         },
         "IDC" : {
            "Unit" : "A",
            "Value" : 0.025438997894525528
         },
         "IDC_2" : {
            "Unit" : "A",
            "Value" : 1.4097353219985962
         },
         "IDC_3" : {
            "Unit" : "A",
            "Value" : null
         },
         "PAC" : {
            "Unit" : "W",
            "Value" : 660.802978515625
         },
         "SAC" : {
            "Unit" : "VA",
            "Value" : 660.84747314453125
         },
         "TOTAL_ENERGY" : {
            "Unit" : "Wh",
            "Value" : 52214.948611111111
         },
         "UAC" : {
            "Unit" : "V",
            "Value" : 236.84129333496094
         },
         "UDC" : {
            "Unit" : "V",
            "Value" : 422.36505126953125
         },
         "UDC_2" : {
            "Unit" : "V",
            "Value" : 484.74273681640625
         },
         "UDC_3" : {
            "Unit" : "V",
            "Value" : null
         },
         "YEAR_ENERGY" : {
            "Unit" : "Wh",
            "Value" : null
         }
      }
   },
   "Head" : {
      "RequestArguments" : {
         "DataCollection" : "CommonInverterData",
         "DeviceId" : "1",
         "Scope" : "Device"
      },
      "Status" : {
         "Code" : 0,
         "Reason" : "",
         "UserMessage" : ""
      },
      "Timestamp" : "2022-06-01T17:36:17+00:00"
   }
}
```

The measurements for the second DC input are revealed in the `UDC_2 `and `IDC_2 `stanza.
I'd assume, that the batteries input values will go into `UDC_3 `and `IDC_3`. Values are still `null` here as the battery is not connected currently.

@jimtng Could you parse the second and third DC input too and provide them as channels in the binding?

_Btw.: The inverters web interface directly shows values for the DC power for each input. Sure, it's simple to calculate the power from the current and voltage given, but if the power of the inputs would be available directly as channels, it would save the effort for additional rules doing the math. Means, **I'd love to get DC power per input as channel** too ;-)_

Let me know, if I can provide more information or if there is something I can test on my side.

Many thanks upfront,
Curlyel

```
openHAB: 3.2.0 release build
Fronius:  Symo Gen24 10.0 Version 1.19.7-1

```

**Comment by jimtng** [comment]
@curlyel please test the jar from here: https://github.com/jimtng/openhab-addons/releases/tag/fronius-extra-dc-0.0.1 and report the result. Thanks.

**Comment by curlyel** [comment]
@jimtng , wow that was ultra fast ;-) Thanks for adding it!
I can confirm that at least the second DC input channels are there now. Can't check the third input yet, as the battery is still not connected:

![image](https://user-images.githubusercontent.com/11849305/171912734-25da542e-18ce-488d-87a8-9481a15adcda.png)

Just a tiny cosmetic flaw: The UDC_3 channel has label "DC Voltage" instead of "DC Voltage 3" what you certainly had in mind certainly  ;-)

![image](https://user-images.githubusercontent.com/11849305/171914854-e3398fac-8e61-4ceb-8511-6c839cf071c2.png)

> I didn't add dc power channels. Not sure if that should be a part of the binding - debatable, I know.

Understand that. Since these values do not come _directly_ from the real device, these would be kind of _synthetic_ channels computed by the binding. This would be only for convenience e.g. of those, who want to plot the power for the different solar generators (e.g. in stacked graph). As this is easily computable by some rule - we certainly can live with it ;-)

Only justification I see, is that some "synthetic" power channels would avoid frequent calling of a power-computation-rule.
So I do agree: debatable ;-)

**Comment by jimtng** [comment]
It's what happens when you're half asleep in bed at 2am!
Please test this one:
https://github.com/jimtng/openhab-addons/releases/tag/fronius-extra-dc-0.0.3

I also added the calculated dc channels.

**Comment by curlyel** [comment]
Sorry for the late reply (was away) ...
> I also added the calculated dc power channels.

Excellent. Thank you very much! Highly appreciated. I Just replaced the jar and linked the items again:

https://github.com/openhab/openhab-addons/issues/12856

I've noticed your pull request on this. Hope, it will get merged even before 3.3.0 release 👍 

Many, many thanks for your help!
